src/Navigation/Navigation.Xamarin.Forms/Service/NavigationService.cache.cs

```diff
@@ -22,20 +23,49 @@ public void ClearCache()
             PageCache.Clear();
         }
 
-        private void CreateCachedPage(
+
+        private static TPage AddOrUpdateContentCache<TPage>(
+            INavigationRegistration registration)
+
+            where TPage : Page
+        {
+            if (PageCache.All(cachedPage => cachedPage.Type != registration.ViewType))
+            {
+                CreateCachedPage(registration.ViewType);
+            }
+
+            var reference = PageCache
+                .First(cachedPage => cachedPage.Type == registration.ViewType)
+                .Reference;
+
+            if (!reference.TryGetTarget(out Page view))
+            {
+                view = GetViewInstance<TPage>(
+                    registration);
+
+                reference.SetTarget(view);
+            }
+
+
+            return (TPage)view;
+        }
+
+
+        private static void CreateCachedPage(
             Type pageType)
         {
-            if (!Configuration.CachePageInstances)
+            if (!Configuration.CacheContent)
             {
                 return;
             }
 
-            if (Configuration.PageTypesToExcludeFromCaching
-                .Contains(pageType))
+            if (Configuration.ContentTypesToExcludeFromCaching?
+                .Contains(pageType) == true)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5575746)